gradle/intTestImage.gradle

```diff
@@ -38,13 +39,48 @@ if (useAllDistribution) {
         from(unpackedPath)
     }
 } else {
-    intTestImage {
-        with project(":distributions").binDistImage
-        into "samples", {
-            from { project(":docs").outputs.samples }
+    configurations { selfRuntime }
+
+    afterEvaluate {
+        if (project.tasks.findByName("jar")) {
+            dependencies { selfRuntime project }
         }
-        doLast { task ->
-            ant.chmod(dir: "$destinationDir/bin", perm: "ugo+rx", includes: "**/*")
+
+        intTestImage {
+            into('bin') {
+                from { project(':launcher').startScripts.outputs.files }
+                fileMode = 0755
+            }
+
+            def runtimeClasspathConfigurations = (rootProject.configurations.coreRuntime
+                    + rootProject.configurations.coreRuntimeExtensions
+                    + project.configurations.selfRuntime)
+            if (project.configurations.findByName('integTestRuntime')) {
+                runtimeClasspathConfigurations += project.configurations.integTestRuntimeClasspath
```

Comment:
You could add a configuration for this and then let the other testing scripts add `extendsFrom` clauses instead of doing if-statements here.

In this script:

```
configurations {
  partialDistribution
}
```

in integrationTest.gradle

```
configurations {
  partialDistribution.extendsFrom(integTestRuntime)
}
```